### PR TITLE
Fix don't allow the domain name when creating the cluster

### LINKED ISSUE #96: Use dns name for nodes
https://github.com/RocksLabs/kvrocks_controller/blob/ef4a9453ea1a0adb84232769b4187ea0813687b0/server/cluster.go#L44

There is restriction - cluster's nodes should set by ip in config. But in k8s env kvrocks more often run as statefulset with static DNS name and dynamic IP address.

So, it would be cool if the controller could use dns names instead if ip addresses.

### PR DESCRIPTION
**Pull request by MateSousa**
This closes #96

### CODE REVIEW

**Comment by git-hulk** [comment]
Hi @MateSousa 

Thanks for your contribution, would you mind transferring your contribute codes to the Apache group? If yes, please help to reply at https://github.com/RocksLabs/kvrocks_controller/issues/94 before merging.

File: util/network.go [hunk 1]
```diff
@@ -45,3 +45,18 @@ func IsPort(port string) bool {
 	}
 	return p > 0 && p < 65536
 }
+
+func IsValidDNS(s string) bool {
```
Comment (git-hulk) [inline util/network.go:49]: IsValidDNS => IsDomainName

File: server/cluster.go [hunk 2]
```diff
@@ -54,7 +54,7 @@ func (req *CreateClusterRequest) validate() error {
 
 	invalidNodes := make([]string, 0)
 	for _, node := range req.Nodes {
-		if !util.IsIPPort(node) {
+		if !util.IsIPPort(node) && !util.IsValidDNS(node) {
```
Comment (git-hulk) [inline server/cluster.go]: How about renaming the IsIPPort to IsHostPort, then checking if the host is an IP or a valid domain name? For checking the domain name, it'd be better to use `net.LookupHost` directly.

Comment (MateSousa) [reply server/cluster.go]: Great ! did it now

File: util/network.go [hunk 3]
```diff
@@ -26,12 +26,13 @@ import (
...
 
-func IsIPPort(s string) bool {
+func IsHostPort(s string) bool {
 	parts := strings.Split(s, ":")
 	if len(parts) != 2 {
 		return false
 	}
-	return IsIP(parts[0]) && IsPort(parts[1])
+
+	return IsIP(parts[0]) && IsPort(parts[1]) || IsDomain(parts[0]) && IsPort(parts[1])
```
Comment (git-hulk) [inline util/network.go]:
```suggestion
	return (IsIP(parts[0]) || IsDomain(parts[0])) && IsPort(parts[1])
```

**Review by git-hulk — APPROVED** [review]

**Pull request merged**